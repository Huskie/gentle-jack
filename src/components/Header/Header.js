import React from 'react';
import styled from 'react-emotion';
import { withTheme } from 'emotion-theming';

import {
  Container,
  FlexContainer,
  Logo,
  Navigation,
} from '..';

const StyledHeader = styled('header')`
    background: ${props => props.theme.colours.alpha};
    color: ${props => props.theme.colours.delta};
    padding: 1.5rem 0;
`;

const BaseHeader = () => (
  <StyledHeader>
    <Container>
      <FlexContainer horizontalAlignment="space-between">
        <Logo label="Gentle Jack" isLink />
        <Navigation />
      </FlexContainer>
    </Container>
  </StyledHeader>
);

BaseHeader.propTypes = {};

BaseHeader.defaultProps = {};


const Header = withTheme(BaseHeader);

export { Header };
