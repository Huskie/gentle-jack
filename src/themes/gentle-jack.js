const theme = {
  borderRadius: '4px',
  colours: {
    alpha: '#2742b8',
    beta: '#222',
    gamma: '#06f',
    delta: '#fff',
    epsilon: '#1b6cff',
    zeta: '#ffd561',
    eta: '#4048ef',
  },
  gradients: {
    alpha: '#4048ef, #2742b8',
  },
  spacing: {
    componentMargin: '16px',
  },
  sizing: {
    maxWidth: '1080px',
  },
};

export { theme };
