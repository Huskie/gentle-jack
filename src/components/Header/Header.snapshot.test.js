import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'emotion-theming';

import { Header } from './Header';
import { theme } from '../../themes/gentle-jack';

describe('Header snapshots:', () => {
  it('component renders with defaults', () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <Header />
      </ThemeProvider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
