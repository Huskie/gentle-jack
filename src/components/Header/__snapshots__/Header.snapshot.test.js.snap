// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshots: component renders with defaults 1`] = `
.emotion-4 {
  background: #4048ef;
  color: #fff;
  padding: 1.5rem 0;
}

.emotion-3 {
  margin: 0 auto;
  max-width: 1080px;
  padding: 0 24px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.emotion-0 {
  font-weight: 700;
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
  text-transform: uppercase;
}

.emotion-0 a {
  color: #fff;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-1 {
  display: none;
  font-size: 0.75rem;
  font-weight: 700;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  text-transform: uppercase;
}

.emotion-1 ul {
  list-style: none;
  margin: 0;
  padding: 0;
}

.emotion-1 li {
  cursor: pointer;
}

<header
  className="emotion-4"
>
  <div
    className="emotion-3"
  >
    <div
      className="emotion-2"
    >
      <div
        className="emotion-0"
      >
        <a
          href="/"
          title="Gentle Jack"
        >
          Gentle Jack
        </a>
      </div>
      <nav
        className="emotion-1"
      >
        <ul>
          <li>
            About
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;
